src/backend/metal/src/command.rs

```diff
@@ -1037,13 +1041,82 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
         // TODO: MTLRenderCommandEncoder.textureBarrier on macOS?
     }
 
-    fn fill_buffer(
+    fn fill_buffer<R>(
         &mut self,
-        _buffer: &native::Buffer,
-        _range: Range<buffer::Offset>,
-        _data: u32,
-    ) {
-        unimplemented!()
+        buffer: &native::Buffer,
+        range: R,
+        data: u32,
+    ) where
+        R: RangeArg<u64>,
+    {
+        let inner = unsafe { &mut *self.inner.get() };
+        let sink = inner.sink.as_mut().unwrap();
+
+        let pipes = self.shared.service_pipes
+            .lock()
+            .unwrap();
+
+        let pso = pipes
+            .get_fill_buffer()
+            .to_owned();
+
+        // TODO: Reuse buffer allocation
+        let fill = self.shared.device.new_buffer_with_data(
+            &data as *const u32 as *const _,
+            mem::size_of::<u32>() as _,
+            metal::MTLResourceOptions::StorageModeShared,
+        );
+
+        inner.retained_buffers.push(fill.clone());
+
+        let start = *range.start().unwrap();
```

Comment:
@kvark Should we assume 0 (i.e. `..` would mean fill buffer) or simply `expect` here?